shared/util.go

```diff
@@ -294,3 +294,23 @@ func CropString(s string, i int) string {
 	}
 	return s[:i]
 }
+
+// GetSharedPath gets the longest path shared between the given paths.
+func GetSharedPath(paths ...string) string {
+	var parts []string
+	for _, path := range paths {
+		if parts == nil {
+			parts = strings.Split(path, "/")
+		} else {
+			otherParts := strings.Split(path, "/")
+			for i, part := range parts {
+				if part == otherParts[i] {
+					continue
+				}
+				parts = append(parts[:i], "")
```

Comment:
Can you commend that this truncates parts to the common bits? Seeing append here makes it first look like something quite different, and interpreting the arguments is a bit of work. Is there not a slice method that would be more obvious?